lib/pbench/agent/results.py

```diff
@@ -2,94 +2,82 @@
-        try:
-            self.result_dir = Path(result_dir).resolve(strict=True)
-        except FileNotFoundError:
-            logger.error("Invalid result directory provided: {}", result_dir)
-            sys.exit(1)
-        else:
-            if not self.result_dir.is_dir():
-                logger.error("Invalid result directory provided: {}", result_dir)
-                sys.exit(1)
-        try:
-            self.target_dir = Path(target_dir).resolve(strict=True)
-        except FileNotFoundError:
-            logger.error("Invalid target directory provided: {}", target_dir)
-            sys.exit(1)
-        else:
-            if not self.target_dir.is_dir():
-                logger.error("Invalid target directory provided: {}", target_dir)
-                sys.exit(1)
-        self.user = user
-        self.prefix = prefix
+        self.result_dir = Path(result_dir)
+        self.target_dir = Path(target_dir)
         self.config = config
         self.logger = logger
 
+    def check_result_target_dir(self, dir_path, dir_name):
+        if not dir_path:
+            raise FileNotFoundError(f"{dir_name} directory not provided")
+        else:
+            try:
+                if dir_name == "Result":
+                    self.result_dir = Path(dir_path).resolve(strict=True)
+                else:
+                    self.target_dir = Path(dir_path).resolve(strict=True)
+            except Exception as exc:
+                raise FileNotFoundError(
+                    "Invalid {} directory provided: {}, ({})", dir_name, dir_path, exc
+                )
+            else:
+                if not Path(dir_path).is_dir():
+                    raise NotADirectoryError(
+                        f"Invalid {dir_name} directory provided: {dir_path}"
+                    )
+
     def make_result_tb(self):
         if os.path.exists(f"{self.result_dir}.copied"):
-            self.logger.debug("Already copied {}", self.result_dir)
-            sys.exit(0)
+            raise Exception("Already copied {}", self.result_dir)
 
         pbench_run_name = self.result_dir.name
         if os.path.exists(f"{self.result_dir}/.running"):
-            self.logger.debug(
+            raise RuntimeError(
                 "The benchmark is still running in {} - skipping; if that is"
                 " not true, rmdir {}/.running, and try again",
                 pbench_run_name,
                 pbench_run_name,
             )
-            sys.exit(0)
 
         mdlog_name = self.result_dir / "metadata.log"
         mdlog = ConfigParser()
         try:
             with mdlog_name.open("r") as fp:
                 mdlog.read_file(fp)
-        except FileNotFoundError:
-            self.logger.debug(
-                "The {}/metadata.log file seems to be missing", pbench_run_name
+        except Exception as exc:
+            raise FileNotFoundError(
+                "The {}/metadata.log file seems to be missing", pbench_run_name, exc
             )
```

Comment:
Again, raising `from exc` would be much more useful than this. And I'm pretty sure that nothing is going to format your template string with your parameters; nor is `FileNotFoundError` going to treat your `exc` argument as a chained "cause" exception.

Reply:
Thank you for pointing out :)

Reply:
Done